lsp-mode.el

```diff
@@ -4462,14 +4470,24 @@ and the position respectively."
                                     filename (error-message-string err)))
                    (file-error (lsp-warn "Failed to process xref entry, file-error, '%s': %s"
                                          filename (error-message-string err)))))))
-    (apply #'append
-           (if (->> locations cl-first lsp-location?)
-               (->> locations
-                    (seq-group-by (-compose #'lsp--uri-to-path #'lsp:location-uri))
-                    (seq-map (-rpartial #'get-xrefs-in-file nil)))
-             (->> locations
-                  (seq-group-by (-compose #'lsp--uri-to-path #'lsp:location-link-target-uri))
-                  (seq-map (-rpartial #'get-xrefs-in-file t)))))))
+
+    (->> locations
+         (seq-sort #'lsp--location-before-p)
+         (seq-group-by (-compose #'lsp--uri-to-path #'lsp--location-uri))
+         (seq-map #'get-xrefs-in-file)
+         (apply #'nconc))))
+
+(defun lsp--location-before-p (a b)
+  "Sort first by file, then by line, then by column."
+  (let ((a-uri (lsp--location-uri a))
+        (b-uri (lsp--location-uri b)))
+    (if (not (string= a-uri b-uri))
+        (string< a-uri b-uri)
+      (-let (((&Range :start (&Position :line a-line :character a-char)) (lsp--location-range a))
```

Comment:
there is `lsp--position-compare`

Reply:
Done. I tweaked a few related things to be clearer.